### PR TITLE
Bug 1862036, Update statement about NAS storage

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1862036

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@jsafrane @dmage @sferich888 PTAL

File: modules/recommended-configurable-storage-technology.adoc [hunk 1]
```diff
@@ -91,10 +91,13 @@ recommended for {product-title} Registry cluster deployment with production work
 
 In a scaled/HA {product-title} registry cluster deployment:
 
-* The preferred storage technology is object storage. The storage technology must support RWX access mode and must ensure read-after-write consistency.
-* File storage and block storage are not recommended for a scaled/HA {product-title} registry cluster deployment with production workloads.
-* All NAS storage are not recommended for {product-title} Registry cluster
-deployment with production workloads.
+* The preferred storage technology is object storage. The storage technology must
+support RWX access mode and must ensure read-after-write consistency.
```
Comment (sferich888) [inline modules/recommended-configurable-storage-technology.adoc:95]: 'read after write' is not a 'storges' [consistency term](https://en.wikipedia.org/wiki/Consistency_model). 

This was a term, that AWS (the service provider) coined with a the [this announcement of AWS S3](https://aws.amazon.com/about-aws/whats-new/2015/08/amazon-s3-introduces-new-usability-enhancements/). 

The point of this is, asking about service, outside of 'object' storage, to meet, have or offer this requirement, is absurd.

File: modules/recommended-configurable-storage-technology.adoc [hunk 2]
```diff
@@ -91,10 +91,13 @@ recommended for {product-title} Registry cluster deployment with production work
...
+* File storage and block storage are not recommended for a scaled/HA
+{product-title} registry cluster deployment with production workloads.
+* NAS storage is not recommended for production workloads on {product-title}
+registry clusters. The storage technology must ensure read-after-write
+consistency.
```
Comment (sferich888) [inline modules/recommended-configurable-storage-technology.adoc:100]: If we're saying this we should require S3 compliant 'object' storage.

**Review by sferich888 — COMMENTED** [review]
I think we need to rediscuss this with PM. This is a very backhanded way of saying we require object storage. Why not just say that's the requirement?

**Comment by sferich888** [comment]
> I think we need to rediscuss this with PM. This is a very backhanded way of saying we require object storage. Why not just say that's the requirement? 

https://en.m.wikipedia.org/wiki/Strong_consistency#:~:text=Strong%20consistency%20is%20one%20of,nodes%2C%20processors%2C%20etc.) May be what your trying to say.

File: modules/recommended-configurable-storage-technology.adoc (hunk 2 above)
Comment (lautou) [reply modules/recommended-configurable-storage-technology.adoc:100]: NAS storage is file storage.
If you say "File storage is not recommended" , so the next sentence (NAS storage is not recommended...) is redundant.
If you want to insist especially on NAS storage, i would turn the first sentence:
"File storage (including NAS storage) and block storage are not recommended..."

File: modules/recommended-configurable-storage-technology.adoc (hunk 1 above)
Comment (lautou) [reply modules/recommended-configurable-storage-technology.adoc:95]: The problem is object storage is available only for customers on cloud.
What about for customers with on-premise installation (vSphere, bare-metal)? File storage is the only option?
So we can't say file storage is not recommended for such customer.
We need a clear guidance, about the storage supportability for the internal registry.

**Comment by ahardin-rh** [comment]
@jsafrane @dmage PTAL

File: modules/recommended-configurable-storage-technology.adoc [hunk 3]
```diff
@@ -84,39 +84,45 @@ In a non-scaled/high-availability (HA) {product-title} registry cluster deployme
...
+Registry, they are not recommended for cluster deployment.
 
 === Scaled registry
 
 In a scaled/HA {product-title} registry cluster deployment:
 
-* The preferred storage technology is object storage. The storage technology must support RWX access mode and must ensure read-after-write consistency.
-* File storage and block storage are not recommended for a scaled/HA {product-title} registry cluster deployment with production workloads.
+* The preferred storage technology is object storage.
+* Object storage must be S3 compliant.
```
Comment (dmage) [inline modules/recommended-configurable-storage-technology.adoc:98]: We support S3, GCS, Azure Blob Storage, Swift. It's strange to ask Azure Blob Storage to be S3 compliant. If you want to use S3 compliant storage, it should be S3 compliant (obviously).

File: modules/recommended-configurable-storage-technology.adoc [hunk 4]
```diff
@@ -84,39 +84,45 @@ In a non-scaled/high-availability (HA) {product-title} registry cluster deployme
...
+* File storage is not recommended for a scaled/HA {product-title} registry cluster
+deployment with production workloads.
+* Block storage is not configurable.
 
 === Metrics
 
 In an {product-title} hosted metrics cluster deployment:
 
 * The preferred storage technology is block storage.
+* Object storage is not configurable.
```
Comment (dmage) [inline modules/recommended-configurable-storage-technology.adoc:109]: What does it mean?

**Comment by dmage** [comment]
one note about S3 compliance, otherwise looks good for the registry part

File: modules/recommended-configurable-storage-technology.adoc [hunk 5]
```diff
@@ -84,39 +84,45 @@ In a non-scaled/high-availability (HA) {product-title} registry cluster deployme
...
+deployment with production workloads.
 ====
 
 === Logging
 
 In an {product-title} hosted logging cluster deployment:
 
 * The preferred storage technology is block storage.
-* It is not recommended to use NAS storage for a hosted metrics cluster deployment with production workloads.
+* It is not recommended to use NAS storage for a hosted metrics cluster deployment
```
Comment (jsafrane) [inline modules/recommended-configurable-storage-technology.adoc:123]: Should it say "File storage is recommended [...]" instead, like in Registry above? NAS can provide block storage too.

File: modules/recommended-configurable-storage-technology.adoc (hunk 3 above)
Comment (jsafrane) [reply modules/recommended-configurable-storage-technology.adoc:98]: +1, list of supported backends (or a link to the list) would be great.

**Comment by ahardin-rh** [comment]
@jsafrane @dmage Thanks for your feedback! Please check behind me to ensure I adequately addressed your concerns. Thanks!

**Comment by ahardin-rh** [comment]
@chao007  Can you please provide QE review, or direct me to the correct contact to do so? Thanks!

**Comment by lautou** [comment]
About S3 compliance, it is one kind of supported storage among object storage.
Think that we support SWIFT object storage for OpenStack platform.
So i would say storage should be S3 compliant or SWIFT compliant.

File: modules/recommended-configurable-storage-technology.adoc (hunk 5 above)
Comment (lautou) [reply modules/recommended-configurable-storage-technology.adoc]: There is a typo in this paragraph about Logging.
Sentence should be:
File storage is not recommended for a hosted **logging** cluster... 
(Not metrics)

**Comment by jsafrane** [comment]
lgtm

File: modules/recommended-configurable-storage-technology.adoc [hunk 6]
```diff
@@ -84,39 +84,47 @@ In a non-scaled/high-availability (HA) {product-title} registry cluster deployme
 
 * The preferred storage technology is object storage followed by block storage. The
 storage technology does not have to support RWX access mode.
-* The storage technology must ensure read-after-write consistency. All NAS storage are not
-recommended for {product-title} Registry cluster deployment with production workloads.
-* While `hostPath` volumes are configurable for a non-scaled/HA {product-title} Registry, they are not recommended for cluster deployment.
+* The storage technology must ensure read-after-write consistency. File storage is
+not recommended for {product-title} Registry cluster deployment with production
```
Comment (codyhoag) [inline modules/recommended-configurable-storage-technology.adoc:88]: For consistency, perhaps `Registry` here should be lower-cased? The other instances of `{product-title} registry cluster deployment` in this doc are all lower-cased.

**Review by codyhoag — COMMENTED** [review]
One nit; everything else LGTM!

**Comment by chao007** [comment]
LGTM
Thanks.

**Pull request merged**